packages/xod-client/src/core/containers/App.jsx

```diff
@@ -45,13 +46,27 @@ export default class App extends React.Component {
   onShowCodeArduino() {
     return R.compose(
       foldEither(
-        R.compose(
-          this.props.actions.addError,
-          R.when(
+        R.cond([
+          [
+            isDeadRefError,
+            err => this.props.actions.addError(
+              deadRefErrorMessage(err),
+              true
```

Comment:
Oh, branching. Again, dead reference is something special. Nope, `App.jsx` should not care.

We need polymorphism at some point anyway, but it would be logical if it happens at the latest point possible. I think type-based decisions should be encapsulated in a module with very simple API. Say, `formatError :: Error -> TheTypeComposeMessageReturns`

In other words, this hunk should look like:

```js
foldEither(
  err => this.props.actions.addError(formatError(err), true),
  this.props.actions.showCode
)
```